Update setup section for std on riscv32imac MCUs (#100)
esp-rs/book: src/installation/index.md

      This target is currently [Tier 2][rust-lang-book--platform-support-tier2]; note the different flavors of `riscv32` target in Rust covering different [`RISC-V` extensions][wiki-riscv-standard-extensions].

    - For `std` applications:

>       Since this target is currently [Tier 3][rust-lang-book--platform-support-tier3], it does not have pre-built objects distributed through `rustup` and, unlike the `no_std` target, **nothing needs to be installed**. Specify the correct target for your device:

Reviewer: ```suggestion
      Since this target is currently [Tier 3][rust-lang-book--platform-support-tier3], it does not have pre-built objects distributed through `rustup` and, unlike the `no_std` target, **nothing needs to be installed**. Here are the available targets:
```

I would avoid using "specify the correct target" as the user might not know how to do that, and, if they use the template, as we recommend, this will be already specified

Author: Done, changed to referencing the rustc book as suggested in your other comment.

      
      - `riscv32imc-esp-espidf` for MCUs which do not support atomics, like ESP32-C2 and ESP32-C3

>       - `riscv32imac-esp-espidf` for MCUs which support atomics, like ESP32-C6, ESP32-H2, and ESP32-P4

Reviewer: Maybe we could include a reference to https://github.com/Nassiel/rust/blob/1f5361b40c4ccfa2f806cc9f293d8399a5651b9a/src/doc/rustc/src/platform-support/esp-idf.md and avoid listing all the MCUs that every target supports

Author: Agree that this book shouldn't be responsible for listing MCUs.
DId you mean to link to the rust book under Nassiel's fork? I assume the master branch of the offical fork is most likely to stay updated.
https://github.com/rust-lang/rust/blob/master/src/doc/rustc/src/platform-support/esp-idf.md
(Any idea why a link like https://doc.rust-lang.org/rustc/platform-support/esp-idf.html doesn't work?)

Reviewer: > DId you mean to link to the rust book under Nassiel's fork? I assume the master branch of the offical fork is most likely to stay updated.

Sorry for the typo, I meant the upstream repo.

> Any idea why a link like https://doc.rust-lang.org/rustc/platform-support/esp-idf.html doesn't work?

Not sure how often they regenerate the mdbook, maybe its every release and the file was recently added, so, maybe not yet deployed.

Author: Done
